integrations/slack/src/middlewares.ts

```diff
@@ -34,12 +44,166 @@ export async function verifySlackRequest(request: Request, { environment }: Slac
     }
 }
 
+const acknowledgeQuery = async ({
+    context,
+    text,
+    userId,
+    channelId,
+    threadId,
+    accessToken,
+    messageType = 'ephemeral',
+}) => {
+    const slackMessageTypes = {
+        ephemeral: 'chat.postEphemeral',
+        permanent: 'chat.postMessage',
+    };
+
+    return slackAPI(
+        context,
+        {
+            method: 'POST',
+            path: slackMessageTypes[messageType], // probably alwasy ephemeral? or otherwise have replies in same thread
+            payload: {
+                channel: channelId,
+                text: `_Asking: ${stripMarkdown(text)}_`,
+                // thread_ts: message.thread_ts,
+                ...(userId ? { user: userId } : {}), // actually shouldn't be optional
+                ...(threadId ? { thread_ts: threadId } : {}),
+            },
+        },
+        {
+            accessToken,
+        }
+    );
+};
+
+/**
+ * We acknowledge the slack request immediately to avoid failures
+ * and "queue" the actual task to be executed in a subsequent request.
+ */
+export async function acknowledgeSlackEvent(req: Request, context: SlackRuntimeContext) {
+    const eventPayload = await parseEventPayload(req);
+    const { type, team_id, text, bot_id, thread_ts, channel, user } = eventPayload.event;
+
+    const { accessToken } = await getInstallationConfig(context, team_id);
+
+    if (!['block_actions'].includes(type)) {
+        if (['message', 'app_mention'].includes(type) && !bot_id) {
+            // check for bot_id so that the bot doesn't trigger itself
+            // stript out the bot-name in the mention and account for user mentions within the query
+            const parsedQuery = stripMarkdown(stripBotName(text, eventPayload.authorizations[0]?.user_id));
+
+            await acknowledgeQuery({
+                context,
+                text: parsedQuery,
+                userId: user,
+                threadId: thread_ts,
+                channelId: channel,
+                accessToken,
+                messageType: 'permanent',
+            });
+        }
+    }
```

Comment:
Can you try having this inside the actual handler and see if it makes any difference in the perceived delay?